cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -517,14 +529,14 @@ chainSyncClientPipelinedWithLedgerState env ledgerState0 validationMode (CSP.Cha
     -- This is where the magic happens. We intercept the blocks and rollbacks
     -- and use it to maintain the correct ledger state.
     goClientStNext
-      :: History (Either Text LedgerStateEvents)
+      :: History (Either LedgerStateError LedgerStateEvents)
       -> Nat n
-      -> CSP.ClientStNext n (BlockInMode CardanoMode, Either Text (LedgerState, [LedgerEvent])) ChainPoint ChainTip m a
-      -> CSP.ClientStNext n (BlockInMode CardanoMode                                          ) ChainPoint ChainTip m a
+      -> CSP.ClientStNext n (BlockInMode CardanoMode, Either LedgerStateError (LedgerState, [LedgerEvent])) ChainPoint ChainTip m a
+      -> CSP.ClientStNext n (BlockInMode CardanoMode                                                      ) ChainPoint ChainTip m a
     goClientStNext history n (CSP.ClientStNext recvMsgRollForward recvMsgRollBackward) = CSP.ClientStNext
       (\blkInMode@(BlockInMode blk@(Block (BlockHeader slotNo _ _) _) _) tip -> let
           newLedgerStateE = case Seq.lookup 0 history of
-            Nothing -> Left "Rolled back too far."
+            Nothing -> Left InvalidRollback
```

Comment:
Would it be useful for debugging to have additional context in `InvalidRollback`? E.g the `slotNo`?